### ISSUE TITLE
ERPNEXT Error - Education Module - Program Enroll Error

### ISSUE THREAD

**Issue by elvirmuslic** [frappe/erpnext#29759]
### Information about bug

Error comes on when enrolling students..

<img width="1072" alt="Screenshot_19" src="https://user-images.githubusercontent.com/56799242/153591374-1bde7ddd-b29e-4ab6-aa91-4f5ec6aaf29f.png">

### Module

education

### Version

	"erpnext": "13.20.0",
	"frappe": "13.20.1"

### Installation method

FrappeCloud

### Relevant log output / Stack trace / Full Error Message.

```shell
### App Versions

{
	"erpnext": "13.20.0",
	"frappe": "13.20.1"
}
```
### Route
```
Form/Program Enrollment/EDU-ENR-2022-00042
```
### Trackeback
```
Syntax error in query:
UPDATE `tabStudent` SET `joining_date`=((datetime.date(2022, 2, 11),),),`modified`=%(param1)s,`modified_by`=%(param2)s WHERE `name` IN (%(param3)s)
Traceback (most recent call last):
  File "apps/frappe/frappe/desk/form/save.py", line 19, in savedocs
    doc.submit()
  File "apps/frappe/frappe/model/document.py", line 939, in submit
    return self._submit()
  File "apps/frappe/frappe/model/document.py", line 928, in _submit
    return self.save()
  File "apps/frappe/frappe/model/document.py", line 287, in save
    return self._save(*args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 341, in _save
    self.run_post_save_methods()
  File "apps/frappe/frappe/model/document.py", line 1007, in run_post_save_methods
    self.run_method("on_submit")
  File "apps/frappe/frappe/model/document.py", line 869, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1161, in composer
    return composed(self, method, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1144, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File "apps/frappe/frappe/model/document.py", line 863, in <lambda>
    fn = lambda self, *args, **kwargs: getattr(self, method)(*args, **kwargs)
  File "apps/erpnext/erpnext/education/doctype/program_enrollment/program_enrollment.py", line 24, in on_submit
    self.update_student_joining_date()
  File "apps/erpnext/erpnext/education/doctype/program_enrollment/program_enrollment.py", line 64, in update_student_joining_date
    frappe.db.set_value("Student", self.student, "joining_date", date)
  File "apps/frappe/frappe/database/database.py", line 703, in set_value
    query.run(debug=debug)
  File "apps/frappe/frappe/query_builder/utils.py", line 59, in execute_query
    return frappe.db.sql(query, params, *args, **kwargs) # nosemgrep
  File "apps/frappe/frappe/database/database.py", line 146, in sql
    self._cursor.execute(query, values)
  File "env/lib/python3.7/site-packages/pymysql/cursors.py", line 148, in execute
    result = self._query(query)
  File "env/lib/python3.7/site-packages/pymysql/cursors.py", line 310, in _query
    conn.query(q)
  File "env/lib/python3.7/site-packages/pymysql/connections.py", line 548, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "env/lib/python3.7/site-packages/pymysql/connections.py", line 775, in _read_query_result
    result.read()
  File "env/lib/python3.7/site-packages/pymysql/connections.py", line 1156, in read
    first_packet = self.connection._read_packet()
  File "env/lib/python3.7/site-packages/pymysql/connections.py", line 725, in _read_packet
    packet.raise_for_error()
  File "env/lib/python3.7/site-packages/pymysql/protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "env/lib/python3.7/site-packages/pymysql/err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.ProgrammingError: (1064, "You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '),),`modified`='2022-02-11 13:11:26.247402',`modified_by`='Administrator' WHE...' at line 1")

Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 68, in application
    response = frappe.api.handle()
  File "apps/frappe/frappe/api.py", line 55, in handle
    return frappe.handler.handle()
  File "apps/frappe/frappe/handler.py", line 31, in handle
    data = execute_cmd(cmd)
  File "apps/frappe/frappe/handler.py", line 68, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "apps/frappe/frappe/__init__.py", line 1214, in call
    return fn(*args, **newargs)
  File "apps/frappe/frappe/desk/form/save.py", line 19, in savedocs
    doc.submit()
  File "apps/frappe/frappe/model/document.py", line 939, in submit
    return self._submit()
  File "apps/frappe/frappe/model/document.py", line 928, in _submit
    return self.save()
  File "apps/frappe/frappe/model/document.py", line 287, in save
    return self._save(*args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 341, in _save
    self.run_post_save_methods()
  File "apps/frappe/frappe/model/document.py", line 1007, in run_post_save_methods
    self.run_method("on_submit")
  File "apps/frappe/frappe/model/document.py", line 869, in run_method
    out = Document.hook(fn)(self, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1161, in composer
    return composed(self, method, *args, **kwargs)
  File "apps/frappe/frappe/model/document.py", line 1144, in runner
    add_to_return_value(self, fn(self, *args, **kwargs))
  File "apps/frappe/frappe/model/document.py", line 863, in <lambda>
    fn = lambda self, *args, **kwargs: getattr(self, method)(*args, **kwargs)
  File "apps/erpnext/erpnext/education/doctype/program_enrollment/program_enrollment.py", line 24, in on_submit
    self.update_student_joining_date()
  File "apps/erpnext/erpnext/education/doctype/program_enrollment/program_enrollment.py", line 64, in update_student_joining_date
    frappe.db.set_value("Student", self.student, "joining_date", date)
  File "apps/frappe/frappe/database/database.py", line 703, in set_value
    query.run(debug=debug)
  File "apps/frappe/frappe/query_builder/utils.py", line 59, in execute_query
    return frappe.db.sql(query, params, *args, **kwargs) # nosemgrep
  File "apps/frappe/frappe/database/database.py", line 146, in sql
    self._cursor.execute(query, values)
  File "env/lib/python3.7/site-packages/pymysql/cursors.py", line 148, in execute
    result = self._query(query)
  File "env/lib/python3.7/site-packages/pymysql/cursors.py", line 310, in _query
    conn.query(q)
  File "env/lib/python3.7/site-packages/pymysql/connections.py", line 548, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "env/lib/python3.7/site-packages/pymysql/connections.py", line 775, in _read_query_result
    result.read()
  File "env/lib/python3.7/site-packages/pymysql/connections.py", line 1156, in read
    first_packet = self.connection._read_packet()
  File "env/lib/python3.7/site-packages/pymysql/connections.py", line 725, in _read_packet
    packet.raise_for_error()
  File "env/lib/python3.7/site-packages/pymysql/protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "env/lib/python3.7/site-packages/pymysql/err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.ProgrammingError: (1064, "You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '),),`modified`='2022-02-11 13:11:26.247402',`modified_by`='Administrator' WHE...' at line 1")

```
### Request Data
```
{
	"type": "POST",
	"args": {
		"doc": "{\"name\":\"EDU-ENR-2022-00041\",\"owner\":\"Administrator\",\"creation\":\"2022-02-11 13:11:24.277197\",\"modified\":\"2022-02-11 13:11:24.277197\",\"modified_by\":\"Administrator\",\"idx\":0,\"docstatus\":0,\"student\":\"EDU-STU-2022-00001\",\"student_name\":\"Matej Kevkić\",\"program\":\"Python\",\"academic_year\":\"2021-2022\",\"enrollment_date\":\"2022-02-11\",\"boarding_student\":0,\"mode_of_transportation\":\"\",\"doctype\":\"Program Enrollment\",\"courses\":[{\"name\":\"fca689c631\",\"owner\":\"Administrator\",\"creation\":\"2022-02-11 13:11:24.277197\",\"modified\":\"2022-02-11 13:11:24.277197\",\"modified_by\":\"Administrator\",\"parent\":\"EDU-ENR-2022-00041\",\"parentfield\":\"courses\",\"parenttype\":\"Program Enrollment\",\"idx\":1,\"docstatus\":0,\"course\":\"Python Advanced\",\"course_name\":\"Python Advanced\",\"doctype\":\"Program Enrollment Course\"},{\"name\":\"149eec4b5d\",\"owner\":\"Administrator\",\"creation\":\"2022-02-11 13:11:24.277197\",\"modified\":\"2022-02-11 13:11:24.277197\",\"modified_by\":\"Administrator\",\"parent\":\"EDU-ENR-2022-00041\",\"parentfield\":\"courses\",\"parenttype\":\"Program Enrollment\",\"idx\":2,\"docstatus\":0,\"course\":\"Python Basics\",\"course_name\":\"Python Basics\",\"doctype\":\"Program Enrollment Course\"},{\"name\":\"05e7f501ad\",\"owner\":\"Administrator\",\"creation\":\"2022-02-11 13:11:24.277197\",\"modified\":\"2022-02-11 13:11:24.277197\",\"modified_by\":\"Administrator\",\"parent\":\"EDU-ENR-2022-00041\",\"parentfield\":\"courses\",\"parenttype\":\"Program Enrollment\",\"idx\":3,\"docstatus\":0,\"course\":\"Python Intermediate\",\"course_name\":\"Python Intermediate\",\"doctype\":\"Program Enrollment Course\"},{\"name\":\"96037c6eac\",\"owner\":\"Administrator\",\"creation\":\"2022-02-11 13:11:24.277197\",\"modified\":\"2022-02-11 13:11:24.277197\",\"modified_by\":\"Administrator\",\"parent\":\"EDU-ENR-2022-00041\",\"parentfield\":\"courses\",\"parenttype\":\"Program Enrollment\",\"idx\":4,\"docstatus\":0,\"course\":\"Typing.com\",\"course_name\":\"Typing.com\",\"doctype\":\"Program Enrollment Course\"}],\"fees\":[]}",
		"action": "Submit"
	},
	"btn": {
		"jQuery224086753011636047611": {
			"events": {
				"click": [
					{
						"type": "click",
						"origType": "click",
						"guid": 2867,
						"namespace": ""
					}
				]
			}
		}
	},
	"freeze": true,
	"headers": {},
	"error_handlers": {},
	"url": "/api/method/frappe.desk.form.save.savedocs"
}
```
### Response Data
```
{
	"exception": "pymysql.err.ProgrammingError: (1064, \"You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '),),`modified`='2022-02-11 13:11:26.247402',`modified_by`='Administrator' WHE...' at line 1\")"
}
```
```

**Comment by elvirmuslic** [comment]
To add to the issue, this issue is also reproducible on a new (_fresh_) setup. 

I have just setup a new bench and website to test it out. **Education module is broken**.

**Comment by paulteach** [comment]
I have the same issue. Has anyone found a solution?

pymysql.err.ProgrammingError: (1064, "You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '),),`modified`='2022-02-14 21:27:45.829505',`modified_by`='xxxx.xxxxx...' at line 1")

**Comment by elvirmuslic** [comment]
Everyone that updated the system has this issue, I've tested on a few
setup. How come there isn't any test setup for this? It's insane..

**Comment by hari-madhavan** [comment]
Hi ,

Here is the fix for the issue . An SQL , which returns a tuple was assigned to a date variable. Unfortuantely not able to create a PR - so attaching the git diff output . if someone else can review and then take it forward it would be great

Regards
Hari
[student_enrollment.txt](https://github.com/frappe/erpnext/files/8103486/student_enrollment.txt)

**Comment by elvirmuslic** [comment]
> Hi ,
> 
> Here is the fix for the issue . An SQL , which returns a tuple was assigned to a date variable. Unfortuantely not able to create a PR - so attaching the git diff output . if someone else can review and then take it forward it would be great
> 
> Regards Hari [student_enrollment.txt](https://github.com/frappe/erpnext/files/8103486/student_enrollment.txt)

There's just no way for us to use a system like this, the issue has been in production for about a week, which is insane when it is software-breaking. It shouldn't be in production at first but if it is, there should be a hotfix like an hour from the issue starting.

**ERPNext**
I contacted support@erpnext.com, I wrote multiple emails, contacted frappe, frappecloud, made a security vulnerability on this platform. Quite some work, yet nothing.. really, really unprofessional, I hope that no one else is dealing with this otherwise their business will fail if it is dependent on this framework/erpnext.

**FrappeCloud**
We use frappecloud and erpnext original repository and that's why we can't change the code.
